census/estimator/trainer/model.py

```diff
@@ -0,0 +1,185 @@
+# Copyright 2016 Google Inc. All Rights Reserved. Licensed under the Apache
+# License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License. You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import multiprocessing
+
+import tensorflow as tf
+from tensorflow.contrib import layers
+from tensorflow.contrib.learn.python.learn.utils import input_fn_utils
+
+tf.logging.set_verbosity(tf.logging.INFO)
+
+CSV_COLUMNS = ['age', 'workclass', 'fnlwgt', 'education', 'education_num',
+           'marital_status', 'occupation', 'relationship', 'race', 'gender',
+           'capital_gain', 'capital_loss', 'hours_per_week', 'native_country',
+           'income_bracket']
+
+LABEL_COLUMN = 'income_bracket'
+
+DEFAULTS = [[0], [''], [0], [''], [0], [''], [''], [''], [''], [''],
+            [0], [0], [0], [''], ['']]
+
+
+INPUT_COLUMNS = [
+    layers.sparse_column_with_keys(
+        column_name='gender', keys=['female', 'male']),
+
+    layers.sparse_column_with_keys(
+        column_name='race',
+        keys=[
+            'Amer-Indian-Eskimo',
+            'Asian-Pac-Islander',
+            'Black',
+            'Other',
+            'White'
+        ]
+    ),
+
+    layers.sparse_column_with_hash_bucket(
+        'education', hash_bucket_size=1000),
+    layers.sparse_column_with_hash_bucket(
+        'marital_status', hash_bucket_size=100),
+    layers.sparse_column_with_hash_bucket(
+        'relationship', hash_bucket_size=100),
+    layers.sparse_column_with_hash_bucket(
+        'workclass', hash_bucket_size=100),
+    layers.sparse_column_with_hash_bucket(
+        'occupation', hash_bucket_size=1000),
+    layers.sparse_column_with_hash_bucket(
+        'native_country', hash_bucket_size=1000),
+
+    # Continuous base columns.
+    layers.real_valued_column('age'),
+    layers.real_valued_column('education_num'),
+    layers.real_valued_column('capital_gain'),
+    layers.real_valued_column('capital_loss'),
+    layers.real_valued_column('hours_per_week'),
+]
+
+
+def build_estimator(model_dir, embedding_size=8, hidden_units=None):
```

Comment:
Explain in comments you are building a wide and deep model, and whats happening here.